aws/s3_test.go

```diff
@@ -33,39 +33,67 @@ func TestMain(m *testing.M) {
 	os.Exit(exitVal)
 }
 
-// TestS3Bucket represents a test S3 bucket.
-type TestS3Bucket struct {
-	name        string
-	tags        []map[string]string
-	isVersioned bool
+// genTestsBucketName generates a test bucket name.
+func genTestBucketName() string {
+	return strings.ToLower("cloud-nuke-test-" + util.UniqueID() + util.UniqueID())
 }
 
-// SetupInfo has test case setup info.
-type SetupInfo struct {
+// createSession creates a new session and returns it.
+func createSession(region string) (*session.Session, error) {
+	if region == "" {
+		var err error
+		region, err = getRandomRegion()
+		if err != nil {
+			return nil, err
+		}
+		logging.Logger.Debugf("Creating session in region - %s", region)
+	}
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	return session, err
+}
+
+// AWSParams has AWS params info.
+type AWSParams struct {
 	region     string
 	awsSession *session.Session
 	svc        *s3.S3
-	bucketName string
 }
 
-// TestNukeS3BucketArgs represents arguments for testNukeS3BucketWrapper
-type TestNukeS3BucketArgs struct {
-	isVersioned       bool
-	checkDeleteMarker bool
-	objectCount       int
-	objectBatchsize   int
+// newAWSParams sets up common operations for nuke S3 tests.
+func newAWSParams() (AWSParams, error) {
+	var params AWSParams
+
+	region, err := getRandomRegion()
+	if err != nil {
+		return params, err
+	}
+	params.region = region
+
+	params.awsSession, err = createSession(region)
+	if err != nil {
+		return params, err
+	}
+
+	params.svc = s3.New(params.awsSession)
+	if err != nil {
+		return params, err
+	}
+
+	return params, nil
 }
 
-// genTestsBucketName generates a test bucket name.
-func genTestBucketName() string {
-	// Call UniqueID twice because even if the nth test tries to reuse a name of the first test
-	// AWS S3 deletion operation might be in progress after nuking.
-	return strings.ToLower("cloud-nuke-test-" + util.UniqueID() + util.UniqueID())
+// TestS3Bucket represents a test S3 bucket
+type TestS3Bucket struct {
+	name        string
+	tags        []map[string]string
```

Comment:
NIT: consider using the structs from aws sdk for type safety.

Reply:
Checked and saw that there is no need for TestS3Bucket struct - easier to have functions which take args and instead of using types and getting into potential conflicts with AWS. `create` and `addObject` have been replaced by `S3TestCreateBucket` and `S3TestBucketAddObject`